### PR TITLE
Add config-dir flag to change .driftctl location

### PR DESCRIPTION
**Pull request by lotoussa**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #555 
| ❓ Documentation  | yes - i must add this flag to the documentation

## Description

I add a flag 'config-dir' to override the .driftctl default path

### CODE REVIEW

File: pkg/cmd/scan.go [hunk 1]
```diff
@@ -153,6 +155,12 @@ func NewScanCmd() *cobra.Command {
 		false,
 		"Includes cloud provider service-linked roles (disabled by default)",
 	)
+	fl.String(
+		"config-dir",
+		"",
+		".driftctl path to use.\n"+
+			"Default is home directory.\n",
+	)
```
Comment (sundowndev) [inline pkg/cmd/scan.go:163]: You can set the default value here directly, using `homedir.Dir()` or `os.Getenv("HOME")`

**Review by sundowndev — CHANGES_REQUESTED** [review]

File: pkg/cmd/scan.go (hunk 1 above)
Comment (lotoussa) [reply pkg/cmd/scan.go:163]: i'm on it, it can be cool to avoid the logic in provider_installer.go

File: pkg/cmd/scan.go [hunk 2]
```diff
@@ -154,6 +157,15 @@ func NewScanCmd() *cobra.Command {
 		"Includes cloud provider service-linked roles (disabled by default)",
 	)
 
+	configDir, _ := homedir.Dir()
+	// Error handled in getProviderDirectory() pkg/terraform/provider_installer.go
```
Comment (sundowndev) [inline pkg/cmd/scan.go:161]: Comment to remove

File: pkg/cmd/scan.go [hunk 3]
```diff
@@ -154,6 +157,15 @@ func NewScanCmd() *cobra.Command {
 		"Includes cloud provider service-linked roles (disabled by default)",
 	)
 
+	configDir, _ := homedir.Dir()
```
Comment (sundowndev) [inline pkg/cmd/scan.go:160]: Are you sure it's safe to ignore the error here?

**Review by sundowndev — DISMISSED** [review]

File: pkg/cmd/scan.go [hunk 4]
```diff
@@ -154,6 +157,14 @@ func NewScanCmd() *cobra.Command {
...
+	fl.String(
+		"config-dir",
+		configDir,
+		".driftctl path to use.\n"+
+			"Default is home directory.\n",
```
Comment (wbeuil) [inline pkg/cmd/scan.go:165]: I find this sentence really confusing.

I'd rather read: 
`Directory path that driftctl uses for configuration`
`Default is the home directory`

**Review by wbeuil — CHANGES_REQUESTED** [review]
What if the `config-dir` is not a directory but a file ?

File: pkg/cmd/scan.go (hunk 4 above)
Comment (sundowndev) [reply pkg/cmd/scan.go:165]: FYI Cobra already handles default values in help message. You shoudn't have to write it manually.

File: pkg/cmd/scan.go (hunk 3 above)
Comment (lotoussa) [reply pkg/cmd/scan.go:160]: Yes the test can demonstrate multiple situations, and if configDir is "" et err != nil, this variable will be handle after that to call tmpDir()

**Comment by lotoussa** [comment]
> What if the `config-dir` is not a directory but a file ?

DEBU[0024] Bound environment variable to flag env=disable_telemetry flag=disable-telemetry value=true
DEBU[0024] New provider library created
DEBU[0024] Starting gRPC client alias=eu-west-3
fork/exec testconfigdir/.driftctl/plugins/darwin_amd64/terraform-provider-aws_v3.19.0: not a directory
Exiting.

File: pkg/cmd/scan.go (hunk 3 above)
Comment (sundowndev) [reply pkg/cmd/scan.go:160]: Not sure to understand, I don't see any fallback to tmpDir() in the diff

Comment (lotoussa) [reply pkg/cmd/scan.go:160]: pkg/terraform/provider_installer.go:78

Comment (wbeuil) [reply pkg/cmd/scan.go:160]: I think what @sundowndev meant is that for the default variable just below `configDir` if `homedir.Dir()` returns an error, we won't have any default displayed inside the `--help` message. Is that it ?

Comment (sundowndev) [reply pkg/cmd/scan.go:160]: Could you put the fallback here instead ? I think it's confusing that it's not close to the definition

**Review by sundowndev — CHANGES_REQUESTED** [review]
Almost good, I think you forgot to remove this statement in `pkg/terraform/provider_installer.go` : 

```go
if p.homeDir == "" {
	p.homeDir = os.TempDir()
}
```

**Review by sundowndev — APPROVED** [review]

**Review by wbeuil — APPROVED** [review]
LGTM

**Review by sundowndev — DISMISSED** [review]

**Review by sundowndev — APPROVED** [review]

**Comment by sjourdan** [comment]
LGTM 

```
❯  DCTL_TF_PROVIDER_VERSION=3.44.0 driftctl-0.10.0 scan --config-dir ./tmp
[...]
❯  tree tmp/.driftctl/
tmp/.driftctl/
└── plugins
    └── linux_amd64
        └── terraform-provider-aws_v3.44.0_x5
```

Just added a companion issue: https://github.com/cloudskiff/driftctl/issues/669

**Pull request merged**